mesonbuild/utils/universal.py

```diff
@@ -2411,22 +2411,20 @@ def is_base(self) -> bool:
         return self.type is OptionType.BASE
 
 
-def pickle_load(filename: str, object_name: str, object_type: T.Type[_PL]) -> _PL:
-    load_fail_msg = f'{object_name} file {filename!r} is corrupted. Try with a fresh build tree.'
+def pickle_load(filename: str, object_name: str, object_type: T.Type[_PL], suggest_reconfigure: bool = True) -> _PL:
+    load_fail_msg = f'{object_name} file {filename!r} is corrupted.'
+    extra_msg = ' Consider reconfiguring the directory with "meson setup --reconfigure".' if suggest_reconfigure else ''
     try:
         with open(filename, 'rb') as f:
             obj = pickle.load(f)
     except (pickle.UnpicklingError, EOFError):
-        raise MesonException(load_fail_msg)
+        raise MesonException(load_fail_msg + extra_msg)
     except (TypeError, ModuleNotFoundError, AttributeError):
         build_dir = os.path.dirname(os.path.dirname(filename))
-        raise MesonException(
-            f"{object_name} file {filename!r} references functions or classes that don't "
-            "exist. This probably means that it was generated with an old "
-            "version of meson. Try running from the source directory "
-            f'meson setup {build_dir} --wipe')
+        m = f"{object_name} file {filename!r} references functions or classes that don't exist."
```

Comment:
Why are we getting rid of "this probably means that [...]"?

Reply:
Also I think it's fine as an inline string rather than saving it to a variable first but ymmv...

Reply:
> Why are we getting rid of "this probably means that [...]"?
> Also I think it's fine as an inline string rather than saving it to a variable first but ymmv...

Hm, oversight on my part because I rephrased those a few times before committing. Updated.